syz-manager/filter.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/google/syzkaller/pkg/cover"
+	"github.com/google/syzkaller/pkg/log"
+)
+
+type CoverFilter struct {
+	enableFilter    bool
+	kcovFilterStart uint32
+	kcovFilterSize  uint32
+	kcovFilterEnd   uint32
+	weightedPCs     map[uint32]float32
+
+	pcsBitmapPath      string
+	targetLittleEndian bool
+	targetArch         string
+}
+
+func (mgr *Manager) initKcovFilter() {
+	covFilterConfig := make(map[string][]string)
+	if err := json.Unmarshal(mgr.cfg.CovFilter, &covFilterConfig); err != nil {
+		log.Logf(0, "no coverage filter is enabled")
+		mgr.kcovFilter.enableFilter = false
+		return
+	}
+	files := covFilterConfig["files"]
```

Comment:
files/functions/pcs should be part of mgrconfig.Config type. No need to parse json manually.